lib/Dialect/Calyx/Export/CalyxEmitter.cpp

```diff
@@ -77,19 +113,38 @@ struct Emitter {
     return op->emitOpError(message);
   }
 
+  /// Helper function for emitting a Calyx section. It emits the body in the
+  /// following format:
+  /// {
+  ///   <body>
+  /// }
+  template <typename Func>
+  void emitCalyxBody(Func emitBody) {
+    os << space() << LBraceEndL();
+    addIndent();
+    emitBody();
+    reduceIndent();
+    indent() << RBraceEndL();
+  }
+
   /// Emits a Calyx section.
   template <typename Func>
   void emitCalyxSection(StringRef sectionName, Func emitBody,
                         StringRef symbolName = "") {
     indent() << sectionName;
     if (!symbolName.empty())
-      os << " " << symbolName;
-    os << " {\n";
-    addIndent();
+      os << space() << symbolName;
+    emitCalyxBody(emitBody);
+  }
 
-    emitBody();
-    reduceIndent();
-    indent() << "}\n";
+  /// Overloaded version for names that require port emission in the section
+  /// name as well, e.g. WhileOp has the following section name:
+  ///      while <port-name> with <group-name> { ... }
+  template <typename Func>
+  void emitCalyxSection(Func emitBody, StringRef symbolName = "") {
```

Comment:
Why have the option of `symbolName` here? In both uses (emitting if- and while control ops), this is unused.

Reply:
Probably not necessary. Will fix.